### PR TITLE
Allow siteless checkout flow to utilize a "redirect_to" url query arg.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
This PR allows the Jetpack siteless "connect-after-checkout" flow to accept and use a `redirect_to` url query arg as the final redirect location after completing a siteless purchase.

## Why are these changes being made?

* Each individual Jetpack product should be able to dictate the users final redirect location after completing a siteless (license-based) purchase. Also there should be a way to dynamically choose the user's final redirect location depending on any specific scenario. This PR allows us to define the final redirect location (in a siteless checkout) by using a `redirect_to` url query parameter.

### Implementation notes:

- This PR includes the basic code changes to allow the "connect-after-checkout" flow (in Calypso checkout; The flow is typically initiated from within the My Jetpack page in the plugin) to accept and utilize a `redirect_to` url query parameter to determine the user's final redirect location after the siteless checkout flow is fully completed. This PR works automatically for most products already because My Jetpack (in the plugin) already includes a `redirect_to` query arg in the checkout link for each product (determined & defined by each product), however Jetpack Stats is an exception and there is some work still to be done to the checkout links for Jetpack Stats because Stats and it's interstitials work differently than the other products. The PR for getting the Stats checkout flow to work well with the `redirect_to` query arg is here: [Automattic/wp-calypso#94831](https://github.com/Automattic/wp-calypso/pull/94831), with the intention to ship **After** this PR is shipped.

## Testing Instructions

(1) Prerequisite:
You need a jetpack site that is either not site or user connected, -or- only site connected and not user connected.
Make sure you are logged into wordpress.com with your A11n account so you can purchase with credits.

**Test "connect-after-checkout" flow:**

- Checkout this branch and `yarn-start`.
- Go to your jetpack site, My Jetpack. Ensure your site is either not user connected, or not site and user connected.
- In My Jetpack, choose any product that allow you to purchase. (Except not Stats or CRM)
- Click into the product interstitial and click to purchase, until you are taken to the Calypso checkout page.
- On checkout, in the URL, replace `https://wordpress.com` with `http://calypso.localhost:3000`
- Also in the url change the &redirect_to= query arg value to something like, https://yoursitedomain.com/wp-admin/admin.php?page=testing_this_redirect, just so you'll definitely identify the final redirect url when you see the final redirect location/link, and you'll so you'll know it worked correctly.
- Go ahead and click the "Complete purchase" button, (using credits is fine). 
- After checkout is completed, your site should auto connect and you should first be redirected to the connection authorization page. Click "Authorize", then you should be redirected to the product licensing thank-you auto activation page. The product license should auto-activate itself.
- Finally once all complete, you should see the "Go to Dashboard" button. Inspect the button and see that the `href` location of the button is the `redirect_to` url you updated in the step above.

**Run Unit Tests:**

- Run: `yarn test-client client/my-sites/checkout/get-thank-you-page-url/test/get-thank-you-page-url.ts`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx [hunk 1]
```diff
@@ -315,31 +319,34 @@ const LicensingActivationThankYou: FC< Props > = ( {
...
-								) }
-							>
-								<span className="licensing-thank-you-auto-activation__dropdown-item-text">
-									{ option.value === 'activate-license-manually' ? (
-										<strong>{ option.label }</strong>
-									) : (
-										option.label
+					{ selectDropdownItems.map( ( option ) => {
+						const { key: itemKey, ...props } = option.props;
+						return (
```
Comment (elliottprogrammer) [inline client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx:324]: This change here is unrelated to applying a `redirect_to` query arg to the siteless checkout flow.
It is merely a fix for a linting warning, `'Missing "key" prop for element in iterator.'`

File: client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx [hunk 2]
```diff
@@ -315,31 +319,34 @@ const LicensingActivationThankYou: FC< Props > = ( {
...
+							<SelectDropdown.Item key={ itemKey } { ...props }>
```
Comment (elliottprogrammer) [inline client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx:325]: This change here is unrelated to applying a `redirect_to` query arg to the siteless checkout flow.
It is merely a fix for a linting warning, `'Missing "key" prop for element in iterator.'`

**Review by kangzj — APPROVED** [review]
Thanks for improving this @elliottprogrammer! Nice one 👍 

I tested purchase for backup and it worked as described. However I found the flow doesn't work for Jetpack Creator plan, where it ends at the following route. 

image

I don't think this is related to the PR because it's happening on production right now as well.
